src/goto-checker/multi_path_symex_only_checker.cpp

```diff
@@ -72,10 +73,14 @@ operator()(propertiest &properties)
 
 void multi_path_symex_only_checkert::generate_equation()
 {
+  // Gather fields for shadow memory instrumentation
+  const auto fields =
+    shadow_memoryt::gather_field_declarations(goto_model, ui_message_handler);
+
   const auto symex_start = std::chrono::steady_clock::now();
 
-  symex_symbol_table =
-    symex.symex_from_entry_point_of(goto_symext::get_goto_function(goto_model));
+  symex_symbol_table = symex.symex_from_entry_point_of(
+    goto_symext::get_goto_function(goto_model), fields);
```

Comment:
Why is it that `symex_from_entry_point_of` cannot just take care of invoking `gather_field_declarations`?

Reply:
Yes, this is a good idea.